src/pages/UserInfo/index.js

```diff
@@ -3,48 +3,179 @@ import logo from '../../images/logo.png';
 import Frame from '../../components/Frame';
 import Button from '../../components/Button';
 import Img from '../../components/Img';
+import MotoInfo from '../../components/MotoInfo';
+import TowInfo from '../../components/TowInfo';
 import { StyledInput, Container } from '../../components/StyledInput/index';
-import { Select, Label } from '../../components/StyledSelect/index';
 import { StyledLink, StyledFieldset } from "./styles";
+import axios from 'axios';
 
 
 class UserInfo extends React.Component {
+
   state = {
     name: '',
-    lastName: '',
     phoneNum:'',
     brand:'',
     cc:'',
     capacity:'',
     plateNum:'',
     type:'',
+    weight:'',
     photo: '',
     vehiPhoto: '',
+    vehicleType:'',
+    edit: false,
   };
 
+  async componentDidMount() {
+    const userType = localStorage.getItem('userType');
+    const email = localStorage.getItem('email');
+    const { data } = await axios({
+      method: 'GET',
+      baseURL:process.env.REACT_APP_SERVER_URL,
+      url: `/${userType}s?email=${email}`
+    })
+
+    if (userType === 'user') {
+      const { name, phoneNum } = data.users[0];
+      this.setState({ vehicleType: 'Moto', name, phoneNum })
+    } else {
+      const { name, phoneNum } = data.suppliers[0];
+      this.setState({ vehicleType: 'Grúa', name, phoneNum })
+    }
+  }
+
+  eraseUser = async(event) => {
+    event.preventDefault();
+    const token = localStorage.getItem('token');
+    const userType = localStorage.getItem('userType');
+
+    await axios({
+      method: 'DELETE',
+      baseURL:process.env.REACT_APP_SERVER_URL,
+      url: `/${userType}s`,
+      headers: {
+        Authorization: `Bearer ${token}`,
+      } 
+    })
+    this.setState({
+      name: '',
+      phoneNum:'',
+    })
+    this.props.history.push(`/login/`);
+    localStorage.clear();
+  };
+
+
   handleChange = (event) => {
     const { name, value } = event.target;
     this.setState({
       [name]: value,
     });
   };
 
-  sendInfo = (event) => {
+  handleChangePlateNum = (param) => {
+    this.setState({
+      plateNum: param
+    })
+  }
+
+  handleChangeCapacity = (param) => {
+    this.setState({
+      capacity: param
+    })
+  }
+
+  handleChangeBrand = (param) => {
+    this.setState({
+      brand: param
+    })
+  }
+
+  handleChangeCyl = (param) => {
+    this.setState({
+      cc: param
+    })
+  }
+
+  handleChangeType = (param) => {
+    this.setState({
+      type: param
+    })
+  }
+
+  handleChangeWeight = (param) => {
+    this.setState({
+      weight: param
+    })
```

Comment:
El mismo estado no se debería modificar y no debería existir en dos componentes al mismo timpo. Veo que el estado se maneja en este componente y adicionalmente se manejan los mismos valores en los componentes de los formularios.
Esto puede causar problemas ya que estamos dejando en las manos del desarrollador (lo que ha de salir mal, saldrá mal) que estos valores se mantengan sincronizados.
El deber ser es: se maneja el estado desde un componente (o global), el estado solo se modifica en ese componente, y los valores se le hacen llegar a los demás por medio de props.

Adicionalmente puedes aprovecharte de la propiedad `name` de `event.target` para solo tener que escribir una función handleChange y no tener que repetir toda la lógica multiples veces

Reply:
Hecho, ya todo se maneja con la función onChange que recibe como parámetro el evento del componente hijo.
Se usa esta misma condición en los componentes hijos.
`onChange={this.handleChange}`